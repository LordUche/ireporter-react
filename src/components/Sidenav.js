import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { Sidebar, Menu, Segment, Icon } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import { logoutUser } from '../redux/actions/auth';

const Sidenav = ({ children, visible, hideSidebar, loggedIn, logout }) => {
  return (
    <Sidebar.Pushable as={Segment}>
      <Sidebar
        as={Menu}
        animation="uncover"
        icon="labeled"
        onHide={hideSidebar}
        vertical
        visible={visible}
        width="thin"
        direction="right"
      >
        <Menu.Item as={Link} to="/" onClick={hideSidebar}>
          <Icon name="home" />
          Home
        </Menu.Item>
        {!loggedIn && (
          <Fragment>
            <Menu.Item
              as={Link}
              to="/login"
              onClick={hideSidebar}
              className="sidebar__login-button"
            >
              <Icon name="sign in" />
              Log In
            </Menu.Item>
            <Menu.Item
              as={Link}
              to="/signup"
              onClick={hideSidebar}
              className="sidebar__signup-button"
            >
              <Icon name="signup" />
              Sign Up
            </Menu.Item>
          </Fragment>
        )}
        {loggedIn && (
          <Fragment>
            <Menu.Item
              as={Link}
              to="/profile"
              onClick={hideSidebar}
              className="sidebar__profile-button"
            >
              <Icon name="user circle" />
              Profile
            </Menu.Item>
            <Menu.Item
              as={Link}
              to="/"
              onClick={() => {
                hideSidebar();
                logout();
              }}
              className="sidebar__logout-button"
            >
              <Icon name="sign out" />
              Log out
            </Menu.Item>
          </Fragment>
        )}
      </Sidebar>

      <Sidebar.Pusher>
        <main>{children}</main>
      </Sidebar.Pusher>
    </Sidebar.Pushable>
  );
};

Sidenav.propTypes = {
  children: PropTypes.node.isRequired,
  visible: PropTypes.bool.isRequired,
  hideSidebar: PropTypes.func.isRequired,
  loggedIn: PropTypes.bool.isRequired,
  logout: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({ loggedIn: state.auth.loggedIn });

export default connect(
  mapStateToProps,
  { logout: logoutUser }
)(Sidenav);
