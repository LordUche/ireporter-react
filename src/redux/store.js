import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { logger } from 'redux-logger';
import thunk from 'redux-thunk';
import rootReducer from './reducers';

const middleware = [thunk];

process.env.NODE_ENV === `development` && middleware.push(logger);

const initialState = { auth: { loggedIn: false, loading: false } };
const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(applyMiddleware(...middleware))
);

export default store;
