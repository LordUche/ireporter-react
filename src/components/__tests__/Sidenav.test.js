import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import Sidenav from '../Sidenav';
import store from '../../redux/store';
import mockRouterOptions from '../../../test/__mocks__/mockRouter';

describe('<Sidenav />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <Sidenav visible hideSidebar={jest.fn()}>
            test
          </Sidenav>
        </Router>
      </Provider>,
      mockRouterOptions
    );
  });

  it('should match snapshot', () => {
    const tree = toJSON(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
