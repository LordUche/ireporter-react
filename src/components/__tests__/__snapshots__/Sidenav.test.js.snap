// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidenav /> should match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "kq1ice",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "kq1ice",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ConnectFunction
        hideSidebar={[MockFunction]}
        visible={true}
      >
        <Sidenav
          hideSidebar={[MockFunction]}
          loggedIn={false}
          logout={[Function]}
          visible={true}
        >
          <SidebarPushable
            as={[Function]}
          >
            <Segment
              className="pushable"
            >
              <div
                className="ui segment pushable"
              >
                <Sidebar
                  animation="uncover"
                  as={[Function]}
                  direction="right"
                  icon="labeled"
                  onHide={[MockFunction]}
                  vertical={true}
                  visible={true}
                  width="thin"
                >
                  <Ref
                    innerRef={
                      Object {
                        "current": <div
                          class="ui vertical labeled icon ui uncover right thin visible sidebar menu"
                        >
                          <a
                            class="item"
                            href="/"
                          >
                            <i
                              aria-hidden="true"
                              class="home icon"
                            />
                            Home
                          </a>
                          <a
                            class="item sidebar__login-button"
                            href="/login"
                          >
                            <i
                              aria-hidden="true"
                              class="sign in icon"
                            />
                            Log In
                          </a>
                          <a
                            class="item sidebar__signup-button"
                            href="/signup"
                          >
                            <i
                              aria-hidden="true"
                              class="signup icon"
                            />
                            Sign Up
                          </a>
                        </div>,
                      }
                    }
                  >
                    <RefFindNode
                      innerRef={
                        Object {
                          "current": <div
                            class="ui vertical labeled icon ui uncover right thin visible sidebar menu"
                          >
                            <a
                              class="item"
                              href="/"
                            >
                              <i
                                aria-hidden="true"
                                class="home icon"
                              />
                              Home
                            </a>
                            <a
                              class="item sidebar__login-button"
                              href="/login"
                            >
                              <i
                                aria-hidden="true"
                                class="sign in icon"
                              />
                              Log In
                            </a>
                            <a
                              class="item sidebar__signup-button"
                              href="/signup"
                            >
                              <i
                                aria-hidden="true"
                                class="signup icon"
                              />
                              Sign Up
                            </a>
                          </div>,
                        }
                      }
                    >
                      <Menu
                        className="ui uncover right thin visible sidebar"
                        icon="labeled"
                        vertical={true}
                      >
                        <div
                          className="ui vertical labeled icon ui uncover right thin visible sidebar menu"
                        >
                          <MenuItem
                            as={[Function]}
                            onClick={[MockFunction]}
                            to="/"
                          >
                            <Link
                              className="item"
                              onClick={[Function]}
                              to="/"
                            >
                              <a
                                className="item"
                                href="/"
                                onClick={[Function]}
                              >
                                <Icon
                                  as="i"
                                  name="home"
                                >
                                  <i
                                    aria-hidden="true"
                                    className="home icon"
                                    onClick={[Function]}
                                  />
                                </Icon>
                                Home
                              </a>
                            </Link>
                          </MenuItem>
                          <MenuItem
                            as={[Function]}
                            className="sidebar__login-button"
                            onClick={[MockFunction]}
                            to="/login"
                          >
                            <Link
                              className="item sidebar__login-button"
                              onClick={[Function]}
                              to="/login"
                            >
                              <a
                                className="item sidebar__login-button"
                                href="/login"
                                onClick={[Function]}
                              >
                                <Icon
                                  as="i"
                                  name="sign in"
                                >
                                  <i
                                    aria-hidden="true"
                                    className="sign in icon"
                                    onClick={[Function]}
                                  />
                                </Icon>
                                Log In
                              </a>
                            </Link>
                          </MenuItem>
                          <MenuItem
                            as={[Function]}
                            className="sidebar__signup-button"
                            onClick={[MockFunction]}
                            to="/signup"
                          >
                            <Link
                              className="item sidebar__signup-button"
                              onClick={[Function]}
                              to="/signup"
                            >
                              <a
                                className="item sidebar__signup-button"
                                href="/signup"
                                onClick={[Function]}
                              >
                                <Icon
                                  as="i"
                                  name="signup"
                                >
                                  <i
                                    aria-hidden="true"
                                    className="signup icon"
                                    onClick={[Function]}
                                  />
                                </Icon>
                                Sign Up
                              </a>
                            </Link>
                          </MenuItem>
                          <EventStack
                            name="click"
                            on={[Function]}
                            pool="default"
                            target="document"
                          />
                        </div>
                      </Menu>
                    </RefFindNode>
                  </Ref>
                </Sidebar>
                <SidebarPusher>
                  <div
                    className="pusher"
                  >
                    <main>
                      test
                    </main>
                  </div>
                </SidebarPusher>
              </div>
            </Segment>
          </SidebarPushable>
        </Sidenav>
      </ConnectFunction>
    </Router>
  </MemoryRouter>
</Provider>
`;
